### DISCUSSION TITLE
Error when including U8glib: "Undefined symbols for architecture x86_64"

### DISCUSSION THREAD

**Discussion by jmauerhan** [bxparks/UnixHostDuino#66] [General]
Hi folks, I'm having a problem when I try to use the U8glib library. In my Makefile I have `ARDUINO_LIBS := U8glib`, and my `.ino` file looks like this:
```
#include <Arduino.h>
#include <U8glib.h>

U8GLIB_SH1106_128X64  My_u8g_Panel(U8G_I2C_OPT_NONE); 
```

When I run make, I get this error:
```
Undefined symbols for architecture x86_64:
  "U8GLIB::initI2C(_u8g_dev_t*, unsigned char)", referenced from:
      U8GLIB::U8GLIB(_u8g_dev_t*, unsigned char) in game.o
  "_u8g_DrawGlyph", referenced from:
      U8GLIB::write(unsigned char) in game.o
  "_u8g_dev_sh1106_128x64_i2c", referenced from:
      U8GLIB_SH1106_128X64::U8GLIB_SH1106_128X64(unsigned char) in game.o
```

I'm very new to Ardunio and EpoxyDuino, so I apologize if I haven't included enough info.

**Comment by bxparks** [comment]
The U8glib library may not be compatible with EpoxyDuino. Many (most?) Arduino libraries do not work with EpoxyDuino without some tweaks. Some will never work EpoxyDuino due to fundamental incompatibilities. I'm not familiar with U8glib so I can't give much more information.

**Reply by jmauerhan** [reply]
That makes sense, I tried a different library for writing to my OLED and it seemed to work fine. 

Thanks for your response!

**Reply by bxparks** [reply]
I was playing around with the U8glib (older) and the U8g2 (newer) libraries. I got a small demo program written for the U8g2 library compiling using the latest EpoxyDuino (v1.5). The older U8glib does not compile under EpoxyDuino.
